x-pack/plugins/code/public/components/main/content.tsx

```diff
@@ -87,6 +93,7 @@ interface Props extends RouteComponentProps<MainRouteParams> {
   onSearchScopeChanged: (s: SearchScope) => void;
   repoScope: string[];
   fetchMoreCommits(repoUri: string): void;
+  currentRepository?: Repository;
```

Comment:
do we really need to introduce a new field just for this change? Can you just issue an add repository filter action and let the general repository filters management logic to handle this case?

Reply:
discussed offline that we still need this field explicitly particular for the case when we move out from the current repository and enter a different repository.